### PR TITLE
fix bugs in KLDivergence

### PR DESCRIPTION
**Pull request by hunto**
## Motivation

1. The computation of KLDivergence Loss is wrong (see definitions in PyTorch docs: https://pytorch.org/docs/master/generated/torch.nn.KLDivLoss.html#torch.nn.KLDivLoss).

## Modification

1. Use the api provided by PyTorch (`torch.nn.functional.kl_div`) to compute the KLDiv.
2. Add a `reduction` parameter. By default, the reduction method is `batchnorm`.
3. Use `F.log_softmax` instead of initializing new `torch.nn.LogSoftmax` module in the `forward` function.
4. Modify the docs.

## BC-breaking (Optional)

No BC-breaking.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmrazor/models/losses/kl_divergence.py [hunk 1]
```diff
@@ -13,35 +13,49 @@ class KLDivergence(nn.Module):
...
+            f'but gets {reduction}.'
+        self.reduction = reduction
+
     def forward(self, preds_S, preds_T):
         """Forward computation.
 
         Args:
             preds_S (torch.Tensor): The student model prediction with
-                shape (N, C, W, H).
+                shape (N, num_classes).
```
Comment (pppppM) [inline mmrazor/models/losses/kl_divergence.py:47]: #29 has been merged. There are some conflicts.
Because the `kl_div` may be used in segmentation, the shape could be (N, C, H, W).

Comment (hunto) [reply mmrazor/models/losses/kl_divergence.py]: Fixed.
